arelle/SocketUtils.py

```diff
@@ -0,0 +1,17 @@
+import socket
+
+INTERNET_CONNECTIVITY = 'internetConnectivity'
+OFFLINE = 'offline'
+
+
+class WarnSocket(socket.socket):
+    """
+    This is a simple wrapper around the socket to print a warning if Arelle attempts to download somthing while running in offline mode.
+    """
+    def __init__(self, family=-1, type=-1, proto=-1, fileno=None):
+        print(f"Arelle is running in offline mode but is attempting a download.")
```

Comment:
We're trying to do this before anything could potentially import and have a reference to the original socket, so that makes logging through the cntrl difficult. We could have a mechanism to inject the logger implementation later, but I'm okay leaving this as is for now